srcs/response/Response.cpp

```diff
@@ -0,0 +1,523 @@
+#include "Response.hpp"
+
+Response::Response(Client client) : _client(client)
+{
+	_path = _client.get_path();
+	_conf = _client.get_conf();
+	std::string tmp = _client.get_request_target();
+	std::size_t found = tmp.find(".");
+	while (found != std::string::npos)
+	{
+		tmp.erase(0, found + 1);
+		_extension = "." + tmp;
+		found = tmp.find(".");
+	}
+	init_code_msg();
+	_status_code = _client.get_status_code();
+	_if_location = _client.get_if_location();
+	if (_if_location)
+		_location = _conf.get_location(_client.get_request_target()).second;
+}
+
+Response::~Response(void)
+{
+}
+
+void Response::set_header_fields(int cont_Leng)
+{
+	std::map<std::string, std::string> headers;
+
+	headers["Content-Length"] = to_String(cont_Leng);
+	if (_status_code == 405)
+	{
+		std::string tmp;
+		if ((_if_location && _location.is_method_allowed("GET") == true) ||
+			(_conf.is_method_allowed("GET") == true))
+			tmp = "GET, ";
+		if ((_if_location && _conf.is_method_allowed("POST") == true) ||
+			(_location.is_method_allowed("POST") == true))
+			tmp += "POST, ";
+		if ((_if_location && _conf.is_method_allowed("DELETE") == true) ||
+			(_location.is_method_allowed("DELETE") == true))
+			tmp += "DELETE, ";
+		tmp = tmp.substr(0, tmp.size() - 2);
+		headers["Allow"] = tmp;
+	}
+	headers["Content-Type"] = content_mime_type(_extension);
```

Comment:
```
	//headers["Content-Type"] = content_mime_type(_extension);
	if (!this->_extension.empty() && this->_extension.compare(".php") &&  this->_extension.compare(".py") )
		headers["Content-Type"] = content_mime_type(this->_extension);
	else
		headers["Content-Type"] = "text/html; charset=utf-8";
```
need to add these codes for cgi to work! (set content-type to "text/html", or else it will just download the page.)

Reply:
Okay nice thanks !